prophy/generators.py

```diff
@@ -29,6 +29,69 @@ def _build_up_implementation(cls):
         """
             Implementation of type creation. To be overridden in derived metaclasses.
         """
+    def __eq__(cls, other):
```

Comment:
Redefining `__eq__` for a metaclass is quite rare and unexpected. Moreover, it muddles the metaclass implementation somewhat further instead of keeping comparison logic in one contiguous piece of code. Consider using free function and documenting what you mean by "comparing" types.